src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -176,80 +176,95 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
         public void DirectoryInfo_Create_NullDirectoryInfo()
         {
             DirectoryInfo info = null;
-            DirectorySecurity security = new DirectorySecurity();
-
-            Assert.Throws<ArgumentNullException>("directoryInfo", () =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+            var security = new DirectorySecurity();
+            Assert.Throws<ArgumentNullException>("directoryInfo", () => DirectoryInfo_Create_Framework(info, security));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NullDirectorySecurity()
         {
-            DirectoryInfo info = new DirectoryInfo("path");
-
-            Assert.Throws<ArgumentNullException>("directorySecurity", () =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, null);
-                }
-                else
-                {
-                    info.Create(null);
-                }
-            });
+            var info = new DirectoryInfo("path");
+            Assert.Throws<ArgumentNullException>("directorySecurity", () => DirectoryInfo_Create_Framework(info, null));
         }
 
         [Fact]
         public void DirectoryInfo_Create_NotFound()
         {
-            using var directory = new TempDirectory();
-            string path = Path.Combine(directory.Path, Guid.NewGuid().ToString(), "ParentDoesNotExist");
-            DirectoryInfo info = new DirectoryInfo(path);
-            DirectorySecurity security = new DirectorySecurity();
+            var directory = new TempAclDirectory();
+            string path = Path.Combine(directory.Path, "ParentDoesNotExist");
+            directory.Dispose(); // Delete parent folder
```

Comment:
Why not use `using` on instantiation instead of manually call Dispose?

Reply:
Because the test is `DirectoryInfo_Create_NotFound`.

Reply:
The idea is to attempt to set security to a file that does not exist.
I wanted to create a valid path first (platform independent, I didn't want to manually create it), then I deleted the directory since I didn't need it.

Reply:
Actually, I should not delete the folder. I'll continue disposing it with using.